src/build_checklist_indicators.Rmd

```diff
@@ -75,57 +75,209 @@ taxon <- trias::gbif_get_taxa(
-print("write here your code, Damiano!")
+taxon %<>% filter(is.finite(nubKey))
+taxon_keys <- taxon %>% select(key) %>% pull(key)
+nub_keys <- taxon %>% distinct(nubKey)
+taxon_backbone_info <-  nub_keys %>%
+  rowwise() %>%
+  do_(interp(~ rgbif::name_usage(key = .$nubKey,
+                         return = "data")))
+taxon_info <- c("key", "species", "genus", "family", "order",
+                "class", "phylum", "kingdom",
+                "rank", "speciesKey", "taxonomicStatus", 
+                "acceptedKey", "accepted")
+taxon_backbone_info %<>%
+  select(key,one_of(taxon_info))
+combined_taxon_info <- taxon %>% left_join(taxon_backbone_info,
+                                           by = c("nubKey" = "key"))
+head(combined_taxon_info)
 ```
 
+### Distribution
+
 Extract distribution information
 
 ```{r retrieve_distribution_information}
-print("write here your code, Damiano!")
+distribution <- data.frame(key = taxon_keys) %>%
+  rowwise() %>%
+  do_(interp(~ as.data.frame(rgbif::name_usage(key = .$key,
+                                               return = "data",
+                                               data = "distribution")))) %>% 
+  select(-source)
+head(distribution)
 ```
 
+### Description
+
 Extract description information
 
 ```{r retrieve_description_information}
-print("write here your code, Damiano!")
+description <- data.frame(key = taxon_keys) %>%
+  rowwise() %>%
+  do_(interp(~ as.data.frame(rgbif::name_usage(key = .$key,
+                                               return = "data",
+                                               data = "description"))))
+description %<>% select(-c(key, language, source))
+description %<>% ungroup()
+head(description, n = 10)
+```
+
+Contrarily to distribution information, description contains a column  `type` with the following descriptors: 
+
+```{r}
+description %>% distinct(type)
+descriptors <- description %>% distinct(type) %>% pull()
+```
+We would like to *tidy* this data.frame, thus having different descriptors as different columns. Unfortunately `dplyr::spread(key, value)` doesn't work due to *duplicates*, i.e. keys with more than one value, for example
+
+```{r duplicates_issue_spread}
+description %>% filter(taxonKey == 141264585 & type == "native range")
+```
+
+A more complex strategy has been found:
+
+```{r tidy_description}
+change_colname <- function(df, new_colname){
+  df %>% 
+    rename(!!new_colname := description) %>%
+    select(-type)
+}
+
+description_tidy <- 
+  purrr::map(descriptors, function(x) description %>% filter(type == x)) %>%
+  purrr::map2(descriptors, ~ change_colname(.x, .y)) %>%
+  purrr::reduce(dplyr::full_join, by = "taxonKey")
+head(description_tidy)
 ```
 
+The chunk code above should be wrapped in a function: Peter Desmet proposed the name `spread_with_duplicates`. It could be put in project package, `trias`, or be part of `dplyr` package eventually. See GitHub issue [#16](https://github.com/trias-project/trias/issues/16).
```

Comment:
@damianooldoni I thought you renamed it to just `spread`? Although, namespace overwriting could be tricky when people should load trias and tidyr, as the order of loading will be important. Maybe keep it `spread_with_duplicates` as long as the features are not part of `tidyr`.

Reply:
Moreover, when stating here : `The chunk code above should be wrapped in a function...` just do it and use the custom spread function. the explanation about the need of the alternative spread implementation is out of scope of this Rmd about data preparation.